Sofa/Component/SolidMechanics/FEM/HyperElastic/src/sofa/component/solidmechanics/fem/hyperelastic/StandardTetrahedralFEMForceField.inl

```diff
@@ -566,6 +566,35 @@ void  StandardTetrahedralFEMForceField<DataTypes>::addKToMatrix(sofa::linearalge
     }
 }
 
+template <class DataTypes>
+void StandardTetrahedralFEMForceField<DataTypes>::buildStiffnessMatrix(core::behavior::StiffnessMatrix* matrix)
+{
+    const sofa::Size nbEdges = m_topology->getNbEdges();
+    const type::vector< Edge>& edgeArray=m_topology->getEdges();
+
+    const edgeInformationVector& edgeInf = edgeInfo.getValue();
+
+    auto dfdx = matrix->getForceDerivativeIn(this->mstate)
+                       .withRespectToPositionsIn(this->mstate);
+
+    for (sofa::Size l = 0; l < nbEdges; ++l)
+    {
+        const auto& einfo = edgeInf[l];
+        const Index node0 = edgeArray[l][0];
+        const Index node1 = edgeArray[l][1];
+        const Index N0 = 3 * node0;
+        const Index N1 = 3 * node1;
+
+        const Matrix3& stiff = einfo.DfDx;
```

Comment:
Just comparing to AddKToMatrix, No need for the kFact ?
`const Matrix3 stiff = einfo.DfDx * (Real)kFact;`

and any idea why there is an offset in the other version?
`const Matrix3 stiff = einfo.DfDx * (Real)kFact;`

Reply:
Yes, one of the goal of the refactoring of the matrix assembly was to get rid of the offset and the k factor. You no longer need them in `buildStiffnessMatrix`. Instead, they are used in the matrix assembly component.

Reply:
ok sorry for the inattentive questions